### PR TITLE
NIAD-790 Added mocked gpc to wiremock for ok and not found status

### PR DESCRIPTION
**Pull request by mmackiewicz1**

### CODE REVIEW

File: docker/wiremock/stubs/__files/emptyPatientDocumentResponse.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+
```
Comment (mattd-kainos) [inline docker/wiremock/stubs/__files/emptyPatientDocumentResponse.json:2]: This should be a FHIR OperationOutcome response

File: docker/wiremock/stubs/__files/emptyPatientDocumentResponse.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (mattd-kainos) [inline docker/wiremock/stubs/__files/emptyPatientDocumentResponse.json:1]: Filename is `emptyPatientDocumentResponse.json` but response is no longer empty

File: docker/wiremock/stubs/__files/emptyPatientDocumentResponse.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "resourceType" : "OperationOutcome",
+  "issue" : [{
```
Comment (mattd-kainos) [inline docker/wiremock/stubs/__files/emptyPatientDocumentResponse.json:3]: This response is very different from the GP Connect API's 404 response

```
{
    "resourceType": "OperationOutcome",
    "meta": {
        "profile": [
            "https://fhir.nhs.uk/StructureDefinition/gpconnect-operationoutcome-1"
        ]
    },
    "issue": [
        {
            "severity": "error",
            "code": "invalid",
            "details": {
                "coding": [
                    {
                        "system": "https://fhir.nhs.uk/ValueSet/gpconnect-error-or-warning-code-1",
                        "code": "NO_RECORD_FOUND",
                        "display": "No Record Found"
                    }
                ]
            },
            "diagnostics": "No record found"
        }
    ]
}
```

https://orange.testlab.nhs.uk/swagger-v1-5-documents.html

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
404 response different from GPC API

**Review by mattd-kainos — APPROVED** [review]
Tested with postman - public demonstrator and wiremock side by side. Only difference is response headers but that's not significant or part of the story. Nice work!

**Review by Broom94 — APPROVED** [review]

File: docker/wiremock/stubs/mappings/retreiveNoneExistingPatientDocument.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "priority": 1,
+  "request": {
+    "method": "GET",
+    "url": "/http://exampleGPSystem.co.uk/GP0001/STU3/1/gpconnect/documents/Binary/non-existing-id"
+  },
+  "response": {
+    "status": 404,
+    "bodyFileName": "notFoundPatientDocumentResponse.json",
+    "headers": {
+      "Content-Type": "application/json"
```
Comment (Broom94) [inline docker/wiremock/stubs/mappings/retreiveNoneExistingPatientDocument.json:11]: content type should be application/fhir+json

File: docker/wiremock/stubs/mappings/retrievePatientDocument.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  "priority": 1,
+  "request": {
+    "method": "GET",
+    "url": "/http://exampleGPSystem.co.uk/GP0001/STU3/1/gpconnect/documents/Binary/07a6483f-732b-461e-86b6-edb665c45510"
+  },
+  "response": {
+    "status": 200,
+    "bodyFileName": "correctPatientDocumentResponse.json",
+    "headers": {
+      "Content-Type": "application/json"
```
Comment (Broom94) [inline docker/wiremock/stubs/mappings/retrievePatientDocument.json:11]: content type should be application/fhir+json

**Review by Broom94 — COMMENTED** [review]
The response returned isn't the same as orangetest labs

**Comment by Broom94** [comment]
There are some missing headers, below is an orangetestlab response header above are what the mock is returning, i'll send you a screenshot

**Pull request merged**